lib/src/pages/layouts/yaru_landscape_layout.dart

```diff
@@ -47,13 +63,38 @@ class YaruLandscapeLayout extends StatefulWidget {
   State<YaruLandscapeLayout> createState() => _YaruLandscapeLayoutState();
 }
 
+const _kLeftPaneResizingRegionWidth = 4.0;
+const _kLeftPaneResizingRegionAnimationDuration = Duration(milliseconds: 250);
+
 class _YaruLandscapeLayoutState extends State<YaruLandscapeLayout> {
   late int _selectedIndex;
+  late double _leftPaneWidth;
+
+  double _initialPaneWidth = 0.0;
+  double _paneWidthMove = 0.0;
+
+  bool _isDragging = false;
+  bool _isHovering = false;
 
   @override
   void initState() {
-    _selectedIndex = widget.selectedIndex;
     super.initState();
+    _selectedIndex = widget.selectedIndex;
+    _leftPaneWidth = widget.leftPaneWidth;
+  }
+
+  @override
+  void didUpdateWidget(covariant YaruLandscapeLayout oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    final width = MediaQuery.of(context).size.width;
+
+    // Avoid left pane to overflow when resizing the window
+    if (widget.allowLeftPaneResize &&
+        _leftPaneWidth >= width - widget.pageMinWidth) {
+      _leftPaneWidth = width - widget.pageMinWidth;
+      widget.onLeftPaneWidthChange?.call(_leftPaneWidth);
+    }
```

Comment:
there's still an assumption that the widget covers the entire window. shouldn't this check be inside the layout builder instead of relying on media query?

Reply:
You're right, I had forgotten this one :)